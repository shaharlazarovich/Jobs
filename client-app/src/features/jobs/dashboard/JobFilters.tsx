import React, { Fragment, useContext } from 'react';
import { Menu, Header } from 'semantic-ui-react';
import { Calendar } from 'react-widgets';
import { RootStoreContext } from '../../../app/stores/rootStore';
import { observer } from 'mobx-react-lite';

const JobFilters = () => {
  const rootStore = useContext(RootStoreContext);
  const { predicate, setPredicate } = rootStore.jobStore;
  return (
    <Fragment data-test="component-job-filters">
      <Menu vertical size={'large'} style={{ width: '100%', marginTop: 50 }}>
        <Header icon={'filter'} attached color={'teal'} content={'Filters'} />
        <Menu.Item
          active={predicate.size === 0}
          onClick={() => setPredicate('all', 'true')}
          color={'blue'}
          name={'all'}
          content={'All Jobs'}
        />
        <Menu.Item
          active={predicate.has('isCreatedByMe')}
          onClick={() => setPredicate('isCreatedByMe', 'true')}
          color={'blue'}
          name={'myjobs'}
          content={"Jobs I Created"}
        />
      </Menu>
      <Header
        icon={'calendar'}
        attached
        color={'teal'}
        content={'Select Date of last job run'}
      />
      <Calendar
        onChange={date => setPredicate('lastRun', date!)}
        value={predicate.get('lastRun') || new Date()}
      />
    </Fragment>
  );
};

export default observer(JobFilters);