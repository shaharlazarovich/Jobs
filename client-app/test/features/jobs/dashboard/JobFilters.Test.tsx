import { ReactWrapper } from "enzyme";
import React from "react";
import { findByTestAttr } from '../../../common/testUtils';
import JobFilters from "../../../../src/features/jobs/dashboard/JobFilters";
import { setupWrapper } from "../../../common/wrapper";
import dateFnsLocalizer from 'react-widgets-date-fns';

dateFnsLocalizer();

describe('Job Filters UniTest', () => {
    let wrapper: ReactWrapper;
    beforeEach(async () => {
        wrapper = await setupWrapper(<JobFilters  />);
    });
    test('renders without error', () => {
        const component = findByTestAttr(wrapper, 'component-job-filters');
        expect(component.length).toBe(1);
    });
    test('should match the snapshot', () => {
        const component = findByTestAttr(wrapper, 'component-job-filters');
        expect(component).toMatchSnapshot();
    });
})