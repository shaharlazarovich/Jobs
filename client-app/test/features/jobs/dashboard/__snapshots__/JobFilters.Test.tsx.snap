// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Filters UniTest should match the snapshot 1`] = `null`;
